src/Npgsql/Internal/Converters/BitStringConverters.cs

```diff
@@ -121,18 +121,10 @@ protected override BitVector32 ReadCore(PgReader reader)
         };
     }
 
-    public override Size GetSize(SizeContext context, BitVector32 value, ref object? writeState)
-        => value.Data is 0 ? 4 : MaxSize;
-
     protected override void WriteCore(PgWriter writer, BitVector32 value)
     {
-        if (value.Data == 0)
```

Comment:
I traced this runtime check all the way back to its introduction, it wasn't introduced to work around some issue on the PG side.